File: client-clojure/src/sirqul_io_t_platform/specs/connection_response.clj
(ns sirqul-io-t-platform.specs.connection-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def connection-response-data
  {
   (ds/opt :connectionId) int?
   (ds/opt :connectionAccountId) string?
   (ds/opt :connectionPendingId) string?
   (ds/opt :display) string?
   (ds/opt :locationDisplay) string?
   (ds/opt :connectionViewedDate) int?
   (ds/opt :profileImage) string?
   (ds/opt :title) string?
   (ds/opt :connectionAccountType) string?
   (ds/opt :trusted) boolean?
   (ds/opt :following) boolean?
   (ds/opt :friendRequested) boolean?
   (ds/opt :friendRequestPending) boolean?
   (ds/opt :blocked) boolean?
   (ds/opt :friend) boolean?
   })

(def connection-response-spec
  (ds/spec
    {:name ::connection-response
     :spec connection-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ConnectionResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ConnectionResponse._

case class ConnectionResponse (
  connectionId: Option[Long],
connectionAccountId: Option[String],
connectionPendingId: Option[String],
display: Option[String],
locationDisplay: Option[String],
connectionViewedDate: Option[Long],
profileImage: Option[String],
title: Option[String],
connectionAccountType: Option[String],
trusted: Option[Boolean],
following: Option[Boolean],
friendRequested: Option[Boolean],
friendRequestPending: Option[Boolean],
blocked: Option[Boolean],
friend: Option[Boolean])

object ConnectionResponse {
  import DateTimeCodecs._

  implicit val ConnectionResponseCodecJson: CodecJson[ConnectionResponse] = CodecJson.derive[ConnectionResponse]
  implicit val ConnectionResponseDecoder: EntityDecoder[ConnectionResponse] = jsonOf[ConnectionResponse]
  implicit val ConnectionResponseEncoder: EntityEncoder[ConnectionResponse] = jsonEncoderOf[ConnectionResponse]
}
